Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py

```diff
@@ -22,6 +22,42 @@
 # plugin architecture.
 #
 
+class TagValueList(list):
+    def __init__(self, tag, epsilon):
+        self.tag = tag
+        self.epsilon = epsilon
+        # Duplicate values for these tags will compared numerically,
+        # instead of checking the values of the dictionary subseriesValues stored as strings.
+        self.subseriesTagsToVectorValues = [
+            "imageOrientationPatient",
+            "diffusionGradientOrientation",
+        ]
+
+    def tagValueToVector(self, value):
+        if value == "":
+            return numpy.array(0)
+        return numpy.array([float(element) for element in value.split("\\")])
+
+    def __contains__(self, value):
+        # The value is a vector, use custom numeric comparison
+        if self.tag in self.subseriesTagsToVectorValues:
+            valueVector = self.tagValueToVector(value)
+            for subseriesValue in self:
+                subseriesVector = self.tagValueToVector(subseriesValue)
+                # The 'eps' value default is set to 1.e-6, which reproduces the same behavior as
+                # the ITK numeric comparison logic (i.e. absolute difference).
+                # Reference:
+                #   Class: ITK/Modules/Numerics/Optimizersv4/include/itkObjectToObjectMetric.hxx
+                #   Method: VerifyDisplacementFieldSizeAndPhysicalSpace
+                #   URL: https://github.com/hinerm/ITK/blob/master/Modules/Numerics/Optimizersv4/include/itkObjectToObjectMetric.hxx#L471-L475.
+                if numpy.allclose(valueVector, subseriesVector, rtol=0.0, atol=self.epsilon):
+                    return True, subseriesValue
+            return False, value
+
+        # The value is a string, use default list __contains__ method
+        value = value.replace(",", "_")  # remove commas so it can be used as an index
```

Comment:
The contains method should not return a modified value. As I originally suggested, I suggest to keep the update of the value on insertion.

Reply:
the issue is that we need to return it for the reason explained in https://github.com/Slicer/Slicer/pull/7362#discussion_r1388277594

I can modify the method `__contains__` of `TagValueList` class  into `insert `or `add ` and move the [append ](https://github.com/Slicer/Slicer/pull/7362/files#diff-ddb2464152946e0fec0e7fcd1ec6cc4aefcd6b8a9ad3ee82a136564525f0c66fR272) into this method; then return the modified value,  **needed** for the [lines](https://github.com/Slicer/Slicer/commit/64b037eb264de0d6f87c4b0aa79e49bd399176ca#diff-ddb2464152946e0fec0e7fcd1ec6cc4aefcd6b8a9ad3ee82a136564525f0c66fL231-R269).

Not sure, the issue is that probably we would need a larger extensive refactoring